repocreation/createRepo.go

```diff
@@ -0,0 +1,212 @@
+func setTeamsAndCollaborators(client *github.Client, ctx context.Context, repoName string, userHandle string) error{
+	addTeamRepoOptions := github.TeamAddTeamRepoOptions{Permission: "admin"}
+	resp, err := client.Teams.AddTeamRepo(ctx, teamID, org, repoName, &addTeamRepoOptions)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Unable to add collaborators"), log.Data{"resp": resp})
+	}
+
+	resp, err = client.Repositories.RemoveCollaborator(ctx, org, repoName, userHandle)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Unable to remove self as a collaborator"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// setBranchProtections sets the protections for both master and develop branches
+func setBranchProtections(client *github.Client, ctx context.Context, repoName string) error {
+	requiredStatusChecks := github.RequiredStatusChecks{
+		Strict:   true,
+		Contexts: []string{},
+	}
+
+	dismissalRestrictionsRequest := github.DismissalRestrictionsRequest{
+		Users: &[]string{},
+		Teams: &[]string{dpTeamSlug},
+	}
+	requiredPullRequestReviewsEnforcementRequest := github.PullRequestReviewsEnforcementRequest{
+		DismissalRestrictionsRequest: &dismissalRestrictionsRequest,
+		DismissStaleReviews:          true,
+		RequireCodeOwnerReviews:      true,
+		RequiredApprovingReviewCount: 1,
+	}
+
+	branchRestrictions := github.BranchRestrictionsRequest{
+		Users: []string{},
+		Teams: []string{dpTeamSlug},
+	}
+
+	protectionRequest := github.ProtectionRequest{
+		RequiredStatusChecks:       &requiredStatusChecks,
+		RequiredPullRequestReviews: &requiredPullRequestReviewsEnforcementRequest,
+		EnforceAdmins:              true,
+		Restrictions:               &branchRestrictions,
+	}
+	_, resp, err := client.Repositories.UpdateBranchProtection(ctx, org, repoName, "master", &protectionRequest)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Update branch protection failed for master"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.UpdateBranchProtection(ctx, org, repoName, "develop", &protectionRequest)
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Update branch protection failed for develop"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "master")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch master"), log.Data{"resp": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "develop")
+	if err != nil {
+		log.ErrorCtx(ctx, errors.Wrap(err, "Adding protection, require signatures failed on branch develop"), log.Data{"resp": resp})
+	}
+	return err
+}
+
+// setDevelopAsDefaultBranch will set the develop branch as the repositories default branch
+func setDevelopAsDefaultBranch(client *github.Client, ctx context.Context, repoName string, repo *github.Repository) error {
```

Comment:
Context should be the first parameter of the function signature - this should apply to all functions in this package that handle context